test/Tests/src/Semantic/Case_Spec.enso

```diff
@@ -163,4 +215,67 @@ spec = Test.group "Pattern Matches" <|
             Cons _ Nil            -> Test.fail "Expected value to match constant."
             _                     -> Test.fail "Expected value to match constant."
 
+    Test.specify "should be able to match on module rather than a type" <|
+        case Vector of
+#            _ : Vector       -> Test.fail "Expected to match on module."
```

Comment:
assume we should clean this up.

Reply:
It should always be a compile time error. Moved to scala tests